src/main/java/com/dell/isg/smi/service/device/discovery/manager/threads/DeviceIdentificationThread.java

```diff
@@ -14,107 +17,121 @@
+		String ipAddress = discoverDeviceInfo.getIpAddress();
+		logger.trace("Started device identification for IP : " + ipAddress);
+		if (IPRangeValidatorUtil.ping(ipAddress)) {
+			discoverDeviceInfo.setMacAddress(
+					IPRangeValidatorUtil.getMacAddressForIp(ipAddress, discoveryDeviceConfigProvider.getArpCommand()));
+			identifyDeviceByIP(ipAddress);
+		} else {
+			discoverDeviceInfo.setDeviceType(DiscoveryDeviceTypeEnum.UNKNOWN.value());
+			discoverDeviceInfo.setStatus(DiscoveryDeviceStatus.NO_DEVICE.name());
+		}
+	}
+
+	private void identifyDeviceByIP(String ipAddress) {
+		if (discoveryDeviceGroupEnum != null
+				&& EnumUtils.isValidEnum(DiscoveryDeviceGroupEnum.class, discoveryDeviceGroupEnum.value())) {
+			amI(ipAddress, discoveryDeviceGroupEnum);
+		} else if (amI(ipAddress, DiscoveryDeviceGroupEnum.SERVER)) {
+			logger.trace(ipAddress + " Identified has Server Type: " + discoverDeviceInfo.getDeviceType());
+		} else if (amI(ipAddress, DiscoveryDeviceGroupEnum.CHASSIS)) {
+			logger.trace(ipAddress + " Identified has Chassis Type: " + discoverDeviceInfo.getDeviceType());
+		} else if (amI(ipAddress, DiscoveryDeviceGroupEnum.SWITCH)) {
+			logger.trace(ipAddress + " Identified has Switch Type: " + discoverDeviceInfo.getDeviceType());
+		} else if (amI(ipAddress, DiscoveryDeviceGroupEnum.STORAGE)) {
+			logger.trace(ipAddress + " Identified has Storage Type: " + discoverDeviceInfo.getDeviceType());
+		} else if (amI(ipAddress, DiscoveryDeviceGroupEnum.IOM)) {
+			logger.trace(ipAddress + " Identified has IOM Type: " + discoverDeviceInfo.getDeviceType());
+		} else if (amI(ipAddress, DiscoveryDeviceGroupEnum.VM)) {
+			logger.trace(ipAddress + " Identified has VM Type: " + discoverDeviceInfo.getDeviceType());
+		} else {
+			discoverDeviceInfo.setDeviceType(DiscoveryDeviceTypeEnum.UNKNOWN.value());
+			discoverDeviceInfo.setStatus(DiscoveryDeviceStatus.UNKNOWN.name());
+		}
+	}
+
+	private boolean amI(String ipAddress, DiscoveryDeviceGroupEnum deviceGroupEnum) {
+		boolean isDeviceIdentified = false;
+		List<DiscoveryRule> discoveryRuleList = discoveryDeviceConfigProvider
+				.getAllDiscoveryRuleByGroup(deviceGroupEnum);
+		for (DiscoveryRule discoveryRule : discoveryRuleList) {
+			String protocol = discoveryRule.getProtocol();
+			String command = discoveryRule.getCommand();
+			List<DeviceType> enabledDeviceTypeList = (List<DeviceType>) CollectionUtils
+					.select(discoveryRule.getDeviceType(), predicateDiscoveryEnabled());
+			if (!CollectionUtils.isEmpty(enabledDeviceTypeList)) {
+				isDeviceIdentified = identify(command, DiscoveryDeviceProtocolEnum.fromValue(protocol),
+						enabledDeviceTypeList);
+			}
+
+		}
+		return isDeviceIdentified;
+	}
+
+	private Predicate<DeviceType> predicateDiscoveryEnabled() {
+		return new Predicate<DeviceType>() {
+			@Override
+			public boolean evaluate(DeviceType deviceType) {
+				if (deviceType == null) {
+					return false;
+				}
+				return deviceType.getEnabled() == Boolean.TRUE;
```

Comment:
return deviceType.getEnabled() should be good enough for this case , you don't need to compare